packages/stack-shared/src/config/format.ts

```diff
@@ -9,7 +9,7 @@ export type Config = {
   [keyOrDotNotation: string]: ConfigValue | undefined,  // must support undefined for optional values
 };
 
-export type NormalizedConfigValue = string | number | boolean | NormalizedConfig | NormalizedConfigValue[];
+export type NormalizedConfigValue = string | number | boolean | null | NormalizedConfig | NormalizedConfigValue[];
```

Comment:
Normalized configs should never contain null

```suggestion
export type NormalizedConfigValue = string | number | boolean | NormalizedConfig | NormalizedConfigValue[];
```